### PR TITLE
Fix: unsubscribe by id

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
Signed-off-by: LaurenceLiZhixin <382673304@qq.com>

# Description

I changed the unsubscribe logic from key level to subscription level, using unique subscribe-id to locate old subscripition.

Please reference the issue this PR will close: #3906

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: configuration/redis/redis.go [hunk 1]
```diff
@@ -258,17 +259,14 @@ func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequ
 	}, nil
 }
 
-func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) error {
+func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
+	subscribeID, _ := uuid.GenerateUUID()
```
Comment (daixiang0) [inline configuration/redis/redis.go:263]: Could we ignore it safely?

File: go.mod [hunk 2]
```diff
@@ -161,7 +161,10 @@ require (
 	github.com/sijms/go-ora/v2 v2.2.22
 )
 
-require github.com/nats-io/nkeys v0.3.0
+require (
+	github.com/hashicorp/go-uuid v1.0.1
```
Comment (daixiang0) [inline go.mod:165]: Could we use `github.com/google/uuid` to reduce deps?

**Comment by yaron2** [comment]
ping @LaurenceLiZhixin

File: go.mod (hunk 2 above)
Comment (LaurenceLiZhixin) [reply go.mod:165]: fine

**Comment by yaron2** [comment]
@daixiang0 please re-review.

File: configuration/requests.go [hunk 3]
```diff
@@ -33,12 +33,13 @@ type SubscribeRequest struct {
 	Metadata map[string]string `json:"metadata"`
 }
 
-// UnSubscribeRequest is the object describing a request to unsubscribe configuration.
-type UnSubscribeRequest struct {
-	Keys []string `json:"keys"`
```
Comment (daixiang0) [inline configuration/requests.go:38]: Does it means we can not use one API to unsubscribe many config?

Comment (LaurenceLiZhixin) [reply configuration/requests.go:38]: One unsubscribe call is only for stopping only on subscription created before. The previous subscription may contains many keys, which means that we can unsubscribe many config once. @daixiang0

Comment (daixiang0) [reply configuration/requests.go:38]: So it is a breaking change for this API?

Comment (LaurenceLiZhixin) [reply configuration/requests.go:38]: Yes, redis is the first impl of this API. @LaurenceLiZhixin

**Review by daixiang0 — APPROVED** [review]

**Pull request merged**